velox/functions/prestosql/MapKeysByTopNValues.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <algorithm>
+#include "velox/functions/prestosql/MapTopN.h"
+
+namespace facebook::velox::functions {
+
+template <typename TExec>
+struct MapKeysByTopNValuesFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExec);
+  void call(
+      out_type<Array<Orderable<T1>>>& out,
+      const arg_type<Map<Orderable<T1>, Orderable<T2>>>& inputMap,
+      int64_t n) {
+    VELOX_USER_CHECK_GE(n, 0, "n must be greater than or equal to 0");
+
+    if (n == 0) {
+      return;
+    }
+
+    using It = typename arg_type<Map<Orderable<T1>, Orderable<T2>>>::Iterator;
+    // This implementation is inspired by the Java version of the
+    // map_keys_by_top_n_values function in Presto:
+    // https://github.com/prestodb/presto/blob/0d8548313fb8ed197d11a4e9ac1257f177364189/presto-main/src/main/java/com/facebook/presto/operator/scalar/sql/MapSqlFunctions.java#L35
+    // The Java code returns: "RETURN IF(n < 0, fail('n must be greater than or
+    // equal to 0'), map_keys(map_top_n(input, n)))"; Here, we utilize the
+    // comparator from MapTopNFunction to sort the input map.
+    using Compare = typename MapTopNFunction<TExec>::template Compare<It>;
+
+    const Compare comparator;
+
+    const size_t ttlSize =
```

Comment:
maybe `containerSize` or just `size`. not sure what does ttl mean in this function.